monix-eval/shared/src/test/scala/monix/eval/TaskStartSuite.scala

```diff
@@ -52,22 +52,23 @@ object TaskStartSuite extends BaseTestSuite {
     assertEquals(f.value, Some(Success(1)))
   }
 
-  testAsync("task.start keeps current Local.Context on join") { _ =>
+  testAsync("task.start shares Local.Context with fibers") { _ =>
     import monix.execution.Scheduler.Implicits.global
     import cats.syntax.all._
     implicit val opts = Task.defaultOptions.enableLocalContextPropagation
 
     val task = for {
       local <- TaskLocal(0)
       _ <- local.write(100)
-      f <- (Task.shift *> local.read <* local.write(200)).start
       v1 <- local.read
+      f <- (Task.shift *> local.read <* local.write(200)).start
+      // Here, before joining, reads are nondeterministic
```

Comment:
Would be possible to have isolation of locals after `start` until `join` happens?

Reply:
I don't think so. If both current and a forked fiber wrote to a Local, which one should win? That's the main question, and I don't think there's a good, generally applicable solution that is also not confusing